jdbc/src/main/java/org/apache/zeppelin/jdbc/SqlCompleter.java

```diff
@@ -0,0 +1,250 @@
+          "/" + metaData.getDriverName().replace(" ", "-").toLowerCase() + "-sql.keywords";
+
+      logger.info("JDBC DriverName:" + driverSpecificKeywords);
+
+      if (SqlCompleter.class.getResource(driverSpecificKeywords) != null) {
+        String driverKeywords =
+            new BufferedReader(new InputStreamReader(
+                SqlCompleter.class.getResourceAsStream(driverSpecificKeywords))).readLine();
+        keywords += "," + driverKeywords.toUpperCase();
+      }
+
+
+      // Add the keywords from the current JDBC connection
+      try {
+        keywords += "," + metaData.getSQLKeywords();
+      } catch (Exception e) {
+        logger.debug("fail to get SQL key words from database metadata: " + e, e);
+      }
+      try {
+        keywords += "," + metaData.getStringFunctions();
+      } catch (Exception e) {
+        logger.debug("fail to get string function names from database metadata: " + e, e);
+      }
+      try {
+        keywords += "," + metaData.getNumericFunctions();
+      } catch (Exception e) {
+        logger.debug("fail to get numeric function names from database metadata: " + e, e);
+      }
+      try {
+        keywords += "," + metaData.getSystemFunctions();
+      } catch (Exception e) {
+        logger.debug("fail to get system function names from database metadata: " + e, e);
+      }
+      try {
+        keywords += "," + metaData.getTimeDateFunctions();
+      } catch (Exception e) {
+        logger.debug("fail to get time date function names from database metadata: " + e, e);
+      }
+
+      // Also allow lower-case versions of all the keywords
+      keywords += "," + keywords.toLowerCase();
+
+    }
+
+    StringTokenizer tok = new StringTokenizer(keywords, ", ");
+    while (tok.hasMoreTokens()) {
+      completions.add(tok.nextToken());
+    }
+
+    return completions;
+  }
+
+  public static Set<String> getDataModelMetadataCompletions(Connection connection)
+      throws SQLException {
+    Set<String> completions = new TreeSet<>();
+    if (null != connection) {
+      getColumnNames(connection.getMetaData(), completions);
+      getSchemaNames(connection.getMetaData(), completions);
+    }
+    return completions;
+  }
+
+  private static void getColumnNames(DatabaseMetaData meta, Set<String> names) throws SQLException {
+
+    try {
+      ResultSet columns = meta.getColumns(meta.getConnection().getCatalog(), null, "%", "%");
+      try {
+
+        while (columns.next()) {
+          // Add the following strings: (1) column name, (2) table name
+          String name = columns.getString("TABLE_NAME");
+          if (!isBlank(name)) {
+            names.add(name);
+            names.add(columns.getString("COLUMN_NAME"));
+            // names.add(columns.getString("TABLE_NAME") + "." + columns.getString("COLUMN_NAME"));
+          }
+        }
+      } finally {
+        columns.close();
+      }
+
+      logger.debug(Joiner.on(',').join(names));
+    } catch (Throwable t) {
```

Comment:
@bzz This is based on Postgresql. Formal author might follow old style under jdk 1.6. I'll change it.